src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/Signatures/SignatureHeader.cs

```diff
@@ -55,7 +56,8 @@ public SignatureKind Kind
             {
                 int callingConventionOrKind = _rawValue & CallingConventionOrKindMask;
 
-                if (callingConventionOrKind <= maxCallingConvention)
+                if (callingConventionOrKind <= maxCallingConvention
+                    && callingConventionOrKind != (int)SignatureCallingConvention.Unmanaged)
```

Comment:
```suggestion
                    || callingConventionOrKind == (int)SignatureCallingConvention.Unmanaged)
```

Reply:
Missed what this was doing.

Reply:
This can use a few tests  `src\libraries\System.Reflection.Metadata\tests\Metadata\Decoding\MethodSignatureTests.cs`